nemo/core/neural_types/elements.py

```diff
@@ -69,28 +70,35 @@ class ElementType(ABC):
     We are relying on Python for inheritance checking"""
 
     def __str__(self):
+        if torch.jit.is_scripting():
+            return "SuppressedForTorchScript"
         return self.__doc__
 
     def __repr__(self):
+        if torch.jit.is_scripting():
+            return "SuppressedForTorchScript"
         return self.__class__.__name__
 
     @property
-    def type_parameters(self) -> Dict:
+    def type_parameters(self) -> Dict[str, Any]:
         """Override this property to parametrize your type. For example, you can specify 'storage' type such as
         float, int, bool with 'dtype' keyword. Another example, is if you want to represent a signal with a
         particular property (say, sample frequency), then you can put sample_freq->value in there.
         When two types are compared their type_parameters must match."""
         return {}
 
     @property
-    def fields(self) -> Optional[Tuple]:
+    def fields(self):
```

Comment:
This should be optional tuple no ?

Reply:
This is the correct type annotation, but it is not supported by TorchScript:
- TorchScript supports only full annotation. `Optional[Tuple]` does not have an underlying type (error in TorchScript)
- TorchScript does not support tuples of an arbitrary size (only lists). Adding underlying type will result in an error in TorchScript, thus I prefer to delete annotation here